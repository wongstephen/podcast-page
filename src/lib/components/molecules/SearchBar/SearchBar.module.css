.container {
  --padding-size: 32px;

  display: flex;
  align-items: center;
  gap: 8px;
  background-color: var(--color-gray-100);

  &:focus-within {
    --border-size: 2px;
    outline: var(--border-size) solid var(--color-green-500);
    outline-offset: calc(-1 * var(--border-size));
  }
}

.input {
  flex: 1;
  width: 100%;
  height: 100%;
  padding: var(--padding-size) 0 var(--padding-size) var(--padding-size);

  &:focus {
    outline: none;
  }
}

.iconContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  padding-right: calc(
    var(--padding-size) + 16px
  ); /* Adjusted for better alignment */
}

.icon {
  color: var(--color-green-500);
}
