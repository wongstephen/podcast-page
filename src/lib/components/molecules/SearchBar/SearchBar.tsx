import React from 'react';
import { Input } from '../../atoms/Input/Input';
import { SearchIcon } from '../../atoms/Icons/Search';
import { cn } from '@/lib/utilities/cn';
import styles from './SearchBar.module.css';

export default function SearchBar({
  className,
  placeholder = 'Search',
  value,
  onChange,
}: React.ComponentProps<'input'>) {
  return (
    <div className={styles.container}>
      <Input
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        className={cn(styles.input, className)}
      />
      <div className={styles.iconContainer}>
        <SearchIcon aria-hidden="true" className={styles.icon} />
      </div>
    </div>
  );
}
