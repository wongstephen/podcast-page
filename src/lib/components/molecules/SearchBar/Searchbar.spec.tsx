import { render } from '@testing-library/react';

import SearchBar from './SearchBar';

describe('SearchBar Component', () => {
  it('renders correctly', () => {
    const { container } = render(<SearchBar />);
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });
});
