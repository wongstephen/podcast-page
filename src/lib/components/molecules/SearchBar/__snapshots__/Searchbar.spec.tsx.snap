// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SearchBar Component matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <input
      class="container input"
      placeholder="Search"
      type="text"
    />
    <div
      class="iconContainer"
    >
      <svg
        aria-hidden="true"
        class="icon"
        fill="none"
        height="24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 19a8 8 0 1 0 0-16 8 8 0 0 0 0 16ZM21 21l-4.35-4.35"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
